watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -113,6 +120,8 @@ def build(include_pretreatment=False):
     # define flowsheet inlets and outlets
     m.fs.feed = FeedZO(property_package=m.fs.prop_nf)
     m.fs.dye_retentate = Product(property_package=m.fs.prop_nf)
+    # TODO: Consider adding a config option to recycle the centrate product back to the inlet
+    m.fs.centrate = Product(property_package=m.fs.prop_nf)
```

Comment:
@MarcusHolly after going through this a bit more, I think we should aim for the same approach we took on the RO w/ERD revisions: include options to `add_dewatering_unit` (in this PR) and perhaps later on we can do the same for RO and get rid of the other NF only flowsheet (th latter shouldn't happen in this PR).

Reply:
In doing so, we can, for example, easily simulate our old results without the dewatering unit and rerun against inclusion of the dewatering unit. We might also want to run cost options for each of the 3 cost methods (centrifuge, belt press filter, plate press filter).

Reply:
We can do the latter by using that new `MultiUnitModelCostingBlock `feature.